chebpy/core/chebtech.py

```diff
@@ -67,8 +68,8 @@ def initfun_adaptive(cls, fun, *, interval=None):
         '''Initialise a Chebtech from the callable fun utilising the adaptive
         constructor to determine the number of degrees of freedom parameter.'''
         interval = interval if interval is not None else prefs.domain
-        hscale = interval2hscale(interval)
-        coeffs = adaptive(cls, fun, hscale=hscale)
+        interval = Interval(*interval)
+        coeffs = adaptive(cls, fun, hscale=interval.hscale)
         return cls(coeffs, interval=interval)
```

Comment:
Ideally, the constructor would be the only place to convert to `Interval`, but we require `.hscale` early.

Reply:
We could pass interval to `adaptive` instead of hscale, this would make the function easier to call on its own?

Reply:
I think it's probably better as is ... no big deal if we have to create an Interval object here just for the purposes of computing an hscale for use downstream. The intention for core.algorithms is that the stuff in there should be as "Chebpy-data-structure" agnostic as possible (so other modules can potentially lift the logice out directly). Which isn't to say that there isn't any dependency in there now. But still wan't to try and limit it. Ultimately I'm persuadable but using Interval here strikes me as the lesser evil.